pkg/apiserver/cmd/start.go

```diff
@@ -0,0 +1,335 @@
+	BearerToken      string
+	PostStartHooks   []PostStartHook
+
+	GetOpenApiDefinition openapi.GetOpenAPIDefinitions
+
+	RestConfig        *rest.Config
+	StorageNamespace  string
+	StorageKubeConfig string
+	Codec             runtime.Codec
+}
+
+type PostStartHook struct {
+	Fn   genericapiserver.PostStartHookFunc
+	Name string
+}
+
+// StartApiServer starts an apiserver.
+func StartApiServer(
+	apis []*builders.APIGroupBuilder,
+	openapidefs openapi.GetOpenAPIDefinitions,
+	title,
+	version string) {
+	logs.InitLogs()
+	defer logs.FlushLogs()
+
+	GetOpenApiDefinition = openapidefs
+
+	signalCh := genericapiserver.SetupSignalHandler()
+	// To disable providers, manually specify the list provided by getKnownProviders()
+	cmd := NewCommandStartServer(apis, signalCh, title, version)
+
+	cmd.Flags().AddFlagSet(pflag.CommandLine)
+	if err := cmd.Execute(); err != nil {
+		panic(err)
+	}
+}
+
+// NewCommandStartServer create a cobra start server command
+func NewCommandStartServer(builders []*builders.APIGroupBuilder, stopCh <-chan struct{}, title, version string) *cobra.Command {
+	o := NewServerOptions(builders)
+
+	// Support overrides
+	cmd := &cobra.Command{
+		Short: "Launch an API server",
+		Long:  "Launch an API server",
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			return o.RunServer(stopCh, title, version)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&o.PrintBearerToken, "print-bearer-token", false,
+		"Print a curl command with the bearer token to test the server")
+	flags.BoolVar(&o.RunDelegatedAuth, "delegated-auth", true,
+		"Setup delegated auth")
+	flags.StringVar(&o.StorageNamespace, "storage-namespace", "default",
+		"Kubernetes namespace that data stored to")
+	flags.StringVar(&o.StorageKubeConfig, "storage-kubeconfig", "",
+		"kubeconfig path used to build the client of kubernetes, use in-cluster config is not specified")
+	o.RecommendedOptions.AddFlags(flags)
+	o.InsecureServingOptions.AddFlags(flags)
+	feature.DefaultFeatureGate.AddFlag(flags)
+
+	klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
+	klog.InitFlags(klogFlags)
+	flags.AddGoFlagSet(klogFlags)
+
+	// Sync the glog and klog flags.
+	klogFlags.VisitAll(func(f *flag.Flag) {
+		goFlag := flag.CommandLine.Lookup(f.Name)
+		if goFlag != nil {
+			goFlag.Value.Set(f.Value.String())
+		}
+	})
+
+	return cmd
+}
+
+// NewServerOptions create a ServerOptions
+func NewServerOptions(b []*builders.APIGroupBuilder) *ServerOptions {
+	versions := []schema.GroupVersion{}
+	for _, b := range b {
+		versions = append(versions, b.GetLegacyCodec()...)
+	}
+
+	codec := builders.Codecs.LegacyCodec(versions...)
+	o := &ServerOptions{
+		RecommendedOptions: genericoptions.NewRecommendedOptions("", codec),
+		APIBuilders:        b,
+		RunDelegatedAuth:   true,
+		Codec:              codec,
+	}
+	o.RecommendedOptions.SecureServing.BindPort = 443
```

Comment:
Why not use [SecureServingOptions](https://github.com/kubernetes/apiserver/blob/fd43ea72cae9a713aae01016b7a58da1bbaf547a/pkg/server/options/serving.go#L37)  instead of a fixed secure port?  just like  `InsecureServingOptions` above.